src/core/server/config/config_service.test.ts

```diff
@@ -108,21 +121,28 @@ test('pushes new config when reloading and config at path has changed', async ()
   expect(valuesReceived).toEqual(['value', 'new value']);
 });
 
-test("throws error if config class does not implement 'schema'", async () => {
-  expect.assertions(1);
-
+test("throws error if 'schema' is not defined for a key", async () => {
   class ExampleClass {}
 
   const config$ = new BehaviorSubject(new ObjectToConfigAdapter({ key: 'value' }));
   const configService = new ConfigService(config$, defaultEnv, logger);
+  configService.setSchema('no-key', schema.string());
 
   const configs = configService.atPath('key', ExampleClass as any);
 
-  try {
-    await configs.pipe(first()).toPromise();
-  } catch (e) {
-    expect(e).toMatchSnapshot();
-  }
+  expect(configs.pipe(first()).toPromise()).rejects.toMatchInlineSnapshot(
```

Comment:
__issue:__ I believe this test will pass even if you change `rejects` to `resolves` since you don't `await` on `expect` :wink: And in test below.

Actually I'm surprised that we don't have any automatic linter-like check for cases like this :/

Reply:
alright, I expected that jest marks test as async if I call `rejects/resolves` in assertion

Reply:
linter rule is not implemented 😞 https://github.com/jest-community/eslint-plugin-jest/issues/54

Reply:
Eh, too bad